dio/lib/src/options.dart

```diff
@@ -396,34 +406,34 @@ class _RequestConfig {
   ///
   /// If you want to receive the response data with  original bytes,
   /// that's to say the type of [Response.data] will be List<int>, use `bytes`
-  ResponseType responseType;
+  ResponseType? responseType;
 
   /// `validateStatus` defines whether the request is successful for a given
   /// HTTP response status code. If `validateStatus` returns `true` ,
   /// the request will be perceived as successful; otherwise, considered as failed.
-  ValidateStatus validateStatus;
+  ValidateStatus? validateStatus;
 
   /// Whether receiving response data when http status code is not successful.
-  bool receiveDataWhenStatusError;
+  bool? receiveDataWhenStatusError;
```

Comment:
`receiveDataWhenStatusError`, `followRedirects` and `maxRedirects` don't need to be nullable.

Reply:
This is a bit tricky as it affect merge-ability. Would not be possible to detect if the source had the bool/int value set or default value of true is being used.

Reply:
I think you are referring to the `mergeOptions` or `merge` methods, right? The thing is, it was impossible to create a `_RequestConfig` with a null value for these fields. [Its constructor used to assign default values if you passed null](https://github.com/flutterchina/dio/pull/1026/commits/d3a208de4bf217663eaa3c83c33cf8d64daaaf56#diff-bed0ea39c9af0b71c65fb305e8c68236e0292cad32dd6693fe8092aefcdfb6a7L360). The way it is now, it would be possible to pass a null value, but my point is it was not possible before. It was only possible to re-set the default value and assign null after the object was created. I just felt that was unlikely to happen, but I obviously don't know how all the usage of `dio` looks like.
I do however think that the original author was confused about where a null value represents a default and where there are non-null default values. I noticed this in code I have written as well, null safety really helps here!